Library/SharedFunctions.swift

```diff
@@ -279,28 +279,8 @@ public func isNativeRiskMessagingControlEnabled() -> Bool {
   return true
 }
 
-public func rewardIsAvailable(project: Project, reward: Reward) -> Bool {
-  let isLimited = reward.isLimitedQuantity
-  let isTimebased = reward.isLimitedTime
-
-  guard isLimited || isTimebased else { return true }
-
-  let remainingQty = rewardLimitRemainingForBacker(project: project, reward: reward)
-  let isRemaining = remainingQty == nil || (remainingQty ?? 0) > 0
-
-  let now = AppEnvironment.current.dateType.init().timeIntervalSince1970
-  let endsAt = reward.endsAt.coalesceWith(now)
-  let timeLimitNotReached = endsAt > now
-
-  // Limited availability is valid if the reward is limited and remaining > 0 OR this reward is not limited.
-  let limitedAvailabilityValid = (isLimited && isRemaining) || !isLimited
-
-  // Timebased availability is valid if the reward is timebased and the time limit has not been reached
-  // OR the reward is not timebased.
-  let timebasedAvailabilityValid = (isTimebased && timeLimitNotReached) || !isTimebased
-
-  // Both types of availability must be valid in order for this reward to be considered available.
-  return limitedAvailabilityValid && timebasedAvailabilityValid
+public func rewardIsAvailable(_ reward: Reward) -> Bool {
+  reward.isAvailable == true || reward.isNoReward
```

Comment:
The no reward ("$1 Pledge without a reward") option is always available. The backend isn't controlling the isAvailable value for these, so I'm forcing it here.

Reply:
What do you think of setting `isAvailable = true` when the no reward reward is created instead? That feels cleaner to me.